src/Orc.SystemInfo/Wmi/WindowsManagementObject.cs

```diff
@@ -64,24 +62,24 @@ internal WindowsManagementObject(IWbemClassObject wmiObject)
             }
         }
 
-        public void Dispose()
+        protected override void DisposeUnmanaged()
         {
-            if (!_disposed)
+            base.DisposeUnmanaged();
+            if(_wbemClassObject is not null)
             {
                 Marshal.ReleaseComObject(_wbemClassObject);
-                _disposed = true;
             }
         }
 
         public IEnumerable<string> GetPropertyNames()
         {
-            ThrowIfDisposed();
+            CheckDisposed();
```

Comment:
this is strange and not clear
what if disposed? will it continue execution?

Reply:
it will throw

Reply:
It's from Catel's Disposable. I don't like naming same as you. It's throwing exception inside, basically do the same as my ThrowIfDisposed(). I think i should make a proposal to Geert to rename it.